DirectEmployers/MyJobs
# [PD-1991] Implement Javascript Lint

# Implement JS Standards and Lint Tool.

* A further task that was folded into this was reducing the bundle sizes.

# Related Tickets

* [PD-1928 NUO](https://directemployersfoundation.atlassian.net/browse/PD-1928) I will take over [@koonsies PR](https://github.com/DirectEmployers/MyJobs/pull/1919)  to get it merged.
* [PD-1836 User Management](https://directemployersfoundation.atlassian.net/browse/PD-1836)  @lemniscate613 will use this new gulpfile and will take care of standards compliance on his own.

# Review Guide

The only user visible change here is smaller JS downloads.

All the non-build Javascript changes are (or should be) to comply with standards or build changes.

## Gulpfile

* There is no longer any need for app specific tasks.
* There are two bundle tasks, one for development (fast, with debugging) and production, which produces much smaller and faster bundles.

# Test Guide

```bash
cd gulp
npm install
gulp
```

* Dynamic reporting at https://secure.my.jobs/reports/view/dynamicoverview should still work as before. As before, only the Contacts report type is active.
